functorch/experimental/_cond.py

```diff
@@ -56,12 +56,22 @@ def trace_cond(proxy_mode, func_overload, pred, true_fn, false_fn, operands):
 
     flat_true_outs, _ = pytree.tree_flatten(true_outs)
     flat_false_outs, _ = pytree.tree_flatten(false_outs)
-    assert(len(flat_true_outs) == len(flat_false_outs))
+    if len(flat_true_outs) != len(flat_false_outs):
+        raise TypeError(
+            f"Expected to return same number of outputs but got:"
+            f"\n  {true_fn.__name__} returns {flat_true_outs}"
```

Comment:
Could you just return the lengths of the outputs? I feel like that would be more informative to the user?

Reply:
Sure.